packages/theme-check-common/src/checks/valid-local-blocks/index.spec.ts

```diff
@@ -0,0 +1,226 @@
+  const theme: MockTheme = {
+    'sections/local-blocks.liquid': `
+          {% schema %}
+          {
+            "name": "Section name",
+            "blocks": [
+              {
+                "type": "local_block",
+                "name": "Local Block"
+              }
+            ],
+            "presets": [
+              {
+                "name": "Default",
+                "blocks": [
+                  {
+                    "id": "static_block",
+                    "type": "static_block",
+                    "static": true
+                  }
+                ]
+              }
+            ]
+          }
+          {% endschema %}
+        `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal(
+    'Sections cannot use static theme blocks together with locally scoped blocks.',
+  );
+});
+
+it('should report errors with correct indicies when sections use static theme blocks together with locally scoped blocks', async () => {
+  const theme: MockTheme = {
+    'sections/local-blocks.liquid': `
+            {% schema %}
+            {
+              "name": "Section name",
+              "blocks": [
+                {
+                  "type": "local_block",
+                  "name": "Local Block"
+                }
+              ],
+              "presets": [
+                {
+                  "name": "Default",
+                  "blocks": [
+                    {
+                      "id": "static_block",
+                      "type": "static_block",
+                      "static": true
+                    }
+                  ]
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+
+  const content = theme['sections/local-blocks.liquid'];
+  const errorContent = content.slice(offenses[0].start.index, offenses[0].end.index);
+  expect(errorContent).to.equal('"static_block"');
+});
+
+it('should report errors when static theme blocks have a name property', async () => {
+  const theme: MockTheme = {
+    'blocks/text.liquid': '',
+    'sections/local-blocks.liquid': `
+      {% schema %}
+      {
+        "name": "Section name",
+        "blocks": [
+          {
+            "type": "@theme"
+          }
+        ],
+        "presets": [
+          {
+            "name": "Default",
+            "blocks": [
+              {
+                "id": "static_block",
+                "type": "static_block",
+                "static": true,
+                "name": "Static Text Block"
+              }
+            ]
+          }
+        ]
+      }
+      {% endschema %}
+    `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal('Static theme blocks cannot have a name property.');
+});
```

Comment:
Fun fact: we recently decided to support preset blocks as a hash:

```
presets: [{
  "name": "My preset",
   "blocks": {
     "my_block_id": { "type": "text" }
   }
}]
```

That means we should ensure the check works in the case of preset blocks as hash, too.

Reply:
Oh my, more complexity 😆